drawBot/context/baseContext.py

```diff
@@ -1905,8 +1905,20 @@ def listNamedInstances(self, fontNameOrPath: SomePath | None = None, fontNumber:
         """
         if fontNameOrPath is None:
             fontNameOrPath = self._font
-        font = getNSFontFromNameOrPath(fontNameOrPath, 10, fontNumber)
-        return variation.getNamedInstancesForFont(font)
+        if not os.path.exists(fontNameOrPath):
+            # the font is installed
+            font = getNSFontFromNameOrPath(fontNameOrPath, 10, fontNumber)
+            # get the url from the font descriptor
+            url = CoreText.CTFontDescriptorCopyAttribute(font.fontDescriptor(), CoreText.kCTFontURLAttribute)
+            if url is None:
+                return dict()
+            else:
+                path = url.path()
+        else:
+            path = fontNameOrPath
+
+        descriptors = getFontDescriptorsFromPath(path)
+        return variation.getNamedInstancesForDescriptors(descriptors)
```

Comment:
We're somehow ignoring fontNumber here.

Yet I don't know what to do, since `CTFontManagerCreateFontDescriptorsFromData()` does not (can not?) distinguish between "multiple instances in a VF", "multiple statics in a TTC" or even "multiple instances for multiple VFs in a TTC".